docs/guide/dev/tips/troubleshooting-exceptions.md

```diff
@@ -0,0 +1,487 @@
+---
+layout: website-normal
+title: Troubleshooting Exceptions and Node Failure
+toc: /guide/toc.json
+---
+
+Whether you're customizing out-of-the-box blueprints, or developing your own custom blueprints, you will
+inevitably have to deal with node failure, or exceptions being thrown by your node. Thankfully Brooklyn
+provides plenty of information to help you locate and resolve any issues you may encounter.
+
+This guide looks at three common failure scenarios and describes the steps that can be taken to
+identify the issue.
+
+## Script failure
+Many blueprints run bash scripts as part of the installation. This section highlights how to identify a problem with
+a bash script.
+
+First let's take a look at the `customize()` method of the Tomcat server blueprint:
```

Comment:
Maybe we want to separate this out into "basic" and "advanced".

The problem with including this chunk of code is that it might look like you need to understand the code in order to investigate a failure.